### PR TITLE
fix(DTFS2-7316): remove transactions from deletion audit logs

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
There is a bug with delete one and delete many audit logs. This is due to us using a sharded database, which is required for change streams. With mongodb 4, we cannot have a sharded database and have transactions that update multiple tables.

## Resolution :heavy_check_mark:
Remove transactions from the deletion helpers

### CODE REVIEW

File: libs/common/src/change-stream/delete-one.test.ts [hunk 1]
```diff
@@ -71,37 +59,23 @@ describe('deleteOne', () => {
...
+        expect(mockDeletionsCollection.insertOne).toHaveBeenCalledWith({
           collectionName: 'users',
-          auditDetails: generateNoUserLoggedInAuditDetails(),
+          deletedDocumentId: documentId,
+          auditRecord: generateMockNoUserLoggedInAuditDatabaseRecord(),
+          expireAt: expect.any(Date) as Date,
         });
-
-        expect(mockUsersCollection.deleteOne).toHaveBeenCalledWith({ _id: { $eq: documentId } }, { session: mockSession });
       });
```
Comment (AlexBramhill) [inline libs/common/src/change-stream/delete-one.test.ts:69]: missing stuff here?

File: portal-api/api-tests/v1/gef/facilities/facilities.delete.api-test.js [hunk 2]
```diff
@@ -91,7 +91,7 @@ describe(baseUrl, () => {
       collectionName: MONGO_DB_COLLECTIONS.FACILITIES,
       auditRecord: expectAnyPortalUserAuditDatabaseRecord(),
       getDeletedDocumentIds: () => facilitiesToDeleteIds,
-      expectedSuccessResponseBody: { acknowledged: true },
+      expectedSuccessResponseBody: { acknowledged: true, deletedCount: 2 },
```
Comment (AlexBramhill) [inline portal-api/api-tests/v1/gef/facilities/facilities.delete.api-test.js:94]: ?

File: portal-api/api-tests/deal-versioning.api-test.js [hunk 3]
```diff
@@ -49,6 +49,7 @@ const generateVersion0ApplicationToSubmit = () => ({
  */
 const generateVersion0ApplicationResponse = (makerId) => ({
   ...generateVersion0ApplicationToSubmit(),
+  mandatoryVersionId: expect.any(Number),
```
Comment (AlexBramhill) [inline portal-api/api-tests/deal-versioning.api-test.js:52]: 👀?

**Review by AlexBramhill — COMMENTED** [review]
few comments but great job getting this through asap

File: portal-api/api-tests/deal-versioning.api-test.js (hunk 3 above)
Comment (MarRobSoftwire) [reply portal-api/api-tests/deal-versioning.api-test.js:52]: The changes I've made have clearly re-ordered the tests a bit & one of them deletes the mandatory criteria. I'll do this properly & add in the mandatory criteria in a beforeAll

File: portal-api/api-tests/v1/gef/facilities/facilities.delete.api-test.js (hunk 2 above)
Comment (MarRobSoftwire) [reply portal-api/api-tests/v1/gef/facilities/facilities.delete.api-test.js:94]: My change to `deleteMany` meant it didnt return the deletedCount, on the grounds of YAGNI. Without transactions this was much easier to do so I've had to add it to the response body here. This was the original behaviour pre-audit logs

**Review by AlexBramhill — APPROVED** [review]

File: libs/common/src/change-stream/delete-one.test.ts [hunk 4]
```diff
@@ -71,37 +59,23 @@ describe('deleteOne', () => {
...
-      it('ends the session', async () => {
-        await deleteOne({
-          db: mockDb,
-          documentId,
-          collectionName: 'users',
-          auditDetails: generateNoUserLoggedInAuditDetails(),
+          expect(mockUsersCollection.deleteOne).toHaveBeenCalledWith({ _id: { $eq: documentId } });
         });
-
-        expect(mockSession.endSession).toHaveBeenCalledTimes(1);
```
Comment (ttbarnes) [inline libs/common/src/change-stream/delete-one.test.ts:104]: Is it worth adding a `toHaveBeenCalledTimes` for `mockUsersCollection.deleteOne`?

**Review by AlexBramhill — DISMISSED** [review]

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Few minor comments to address please.

**Review by ttbarnes — DISMISSED** [review]
LGTM excluding other's reviews

File: libs/common/src/change-stream/delete-one.ts [hunk 5]
```diff
@@ -67,14 +57,12 @@ const deleteDocumentWithAuditLogs = async ({ documentId, collectionName, db, aud
  */
 export const deleteOne = async ({ documentId, collectionName, db, auditDetails }: DeleteOneParams): Promise<DeleteResult> => {
   if (process.env.CHANGE_STREAM_ENABLED === 'true') {
-    await deleteDocumentWithAuditLogs({
+    return await deleteDocumentWithAuditLogs({
```
[the remark this answers is not in the document]
Comment (JoshBinns2000) [reply libs/common/src/change-stream/delete-one.ts:60]: `return await` is redundant as the return value of this function will be a `Promise` independent on whether or not you `await` the final call - this is more of a semantic thing as `return await` makes it clear to the reader that the final function being called is itself `async`. Take the two functions below as an example

```typescript
const collection = await mongoDbClient.getCollection(...);

const deleteOneByIdVersion1 = async (id: ObjectId): Promise<boolean> => {
  try {
    await collection.deleteOne({ _id: { $eq: id } });
    return true; // return value is not itself a 'Promise', so no "return await"
  } catch (error) {
    console.error(error);
    return false; // return value is not itself a 'Promise', so no "return await"
  }
};

const deleteOneByIdVersion2 = async (id: ObjectId): Promise<DeleteResult | boolean> => {
  try {
    return await collection.deleteOne({ _id: { $eq: id } }); // return value is a 'Promise', so "return await"
  } catch (error) {
    console.error(error);
    return false; // return value is not itself a 'Promise', so no "return await"
  }
};
```

**Review by abhi-markan — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**